src/Sylius/Component/Promotion/Repository/PromotionCouponRepositoryInterface.php

```diff
@@ -32,4 +33,12 @@ public function createQueryBuilderByPromotionId($promotionId);
      * @return int
      */
     public function countByCodeLength($codeLength);
+
+    /**
+     * @param string $promotionCode
+     * @param string $code
+     *
+     * @return PromotionInterface
+     */
+    public function findOneByPromotionCodeAndCode($promotionCode, $code);
```

Comment:
@TheMadeleine of course your proposal looks better, the only thing I say, it can be confusing to have path
`/promotions/{promotionCode}/coupons/{code}` and then in routing `[$code, $promotionCode]`.